.github/workflows/formatting.yml

```diff
@@ -21,7 +21,7 @@ jobs:
           path: |
             ~/.cargo/registry
             ~/.cargo/git
-          key: ${{ runner.os }}-format-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+          key: clippy
```

Comment:
I don't understand what this does — if the key is a constant, doesn't it mean the cache will never be invalidated?

Reply:
So... We don't have an os matrix here. And we only test with one Rust. If anybody changes any `Cargo.lock` the cache is invalid and will compile everything again.

Do we want to invalidate the cache of crates?

Reply:
Will that happen automatically when the dependencies change? I assume so, but I'm not sure.

Reply:
Yeah then we of course recompile dependencies and add new ones. I'd like to find a way to just not always compile everything for clippy. The run is 14 minutes now which is too much.

When we hit certain limits github will anyhow delete our caches and we get it invalidated.